Cabal-syntax/src/Distribution/Types/DependencyMap.hs

```diff
@@ -54,6 +64,7 @@ constrainBy
 constrainBy = foldl' tightenConstraint
   where
     tightenConstraint (DependencyMap l) (PackageVersionConstraint pn vr) = DependencyMap $
-      case Map.lookup pn l of
+      -- TODO: Need to establish where these constraints come from
+      case Map.lookup (pn, Public) l of
         Nothing -> l
-        Just (vr', cs) -> Map.insert pn (intersectVersionRanges vr' vr, cs) l
+        Just (vr', cs) -> Map.insert (pn, Public) (intersectVersionRanges vr' vr, cs) l
```

Comment:
I think this is wrong -- and tracing upwards where this function is called and where the arguments are from will unveil potential sources of bugs. 

For example, look at `finalizePD` and `combinedConstraints` where `allConstraints` originates, the private scope is just ignored when those constraints are created.

Reply:
Indeed. I've fixed this by propagating the constraints from `allConstraints` up.